src/OpenSage.Game/Logic/Orders/OrderProcessor.cs

```diff
@@ -93,7 +93,25 @@ public void Process(IEnumerable<Order> orders)
 
                             foreach (var unit in player.SelectedUnits)
                             {
-                                unit.ProductionUpdate.QueueProduction(objectDefinition);
+                                // Only units that can produce stuff should produce it
+                                unit?.ProductionUpdate.QueueProduction(objectDefinition);
```

Comment:
This crashed when having standard units inside the selection

Reply:
When would `unit` be null? I feel like we should filter null objects out before this.